src/orange/OrangeTypes.hh

```diff
@@ -436,6 +424,67 @@ CELER_CONSTEXPR_FUNCTION Sense to_sense(SignedSense s)
     return Sense(static_cast<int>(s) >= 0);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Wrapper for a sense value which associated metadata.
+ */
+class SenseMeta
```

Comment:
At this point we should put SenseMeta (and related Sense helper functions) into its own header 😄